inst/EpiModel2.rmd

```diff
@@ -0,0 +1,684 @@
+
+```{r echo = TRUE, eval = FALSE}
+init <- init.net(i.num = 50, i.num.m2 = 50)
+param <- param.net(inf.prob = 0.1, inf.prob.m2 = 0.1,
+                   rec.rate = 0.02, rec.rate.m2 = 0.02,
+                   act.rate = 5)
+```
+
+should be updated to
+
+```{r echo = TRUE, eval = FALSE}
+init <- init.net(i.num = 50, i.num.g2 = 50)
+param <- param.net(inf.prob = 0.1, inf.prob.g2 = 0.1,
+                   rec.rate = 0.02, rec.rate.g2 = 0.02,
+                   act.rate = 5)
+```
+
+## Using accessor functions
+
+The new asseccor functions are easy to use utilities for getting, setting and appending
+data, replacing `$`  and `[[ ]]` operations.
+
+```{r echo = TRUE, eval = FALSE}
+dat <- list()
+partners <- rpois(100, 1)
+dat$attr$partners <- partners
+```
+
+or 
+
+```{r echo = TRUE, eval = FALSE}
+dat <- list()
+partners <- rpois(100, 1)
+dat[["attr"]][["partners"]] <- partners
+```
+
+should be updated in script to:
+
+```{r echo = TRUE, eval = FALSE}
+dat <- lisit()
+partners <- rpois(100, 1)
+dat <- set_attr(dat, "partners", partners)
+```
+
+Similarly for retrieval and setting of list elements.
+
+```{r echo = TRUE, eval = FALSE}
+partners <- dat$attr$partners
+```
+
+or 
+
+```{r echo = TRUE, eval = FALSE}
+partners <- dat[["attr"]][["partners"]]
+```
+
+should be updated to
+
+```{r echo = TRUE, eval = FALSE}
+partners <- get_attr(dat, "partners")
+```
+
+And
+
+```{r echo = TRUE, eval = FALSE}
+newPartners <- rpois(2, 1)
+dat$attr$partners[101:102] <- newPartners
+```
+
+or 
+
+```{r echo = TRUE, eval = FALSE}
+newPartners <- rpois(2, 1)
+dat[["attr"]][["partners"]][101:102] <- newPartners
+```
+
+should be updated to
+
+```{r echo = TRUE, eval = FALSE}
+newPartners <- rpois(2, 1)
+dat <- append_attr(dat, "partners", newPartners, 2)
+```
+
+See `help("dat_get_set", "EpiModel")` for the complete list of accessor functions and their use.
+
+# Worked Examples
+
+The following section presents three worked examples of models built in pre-EpiModel 2.0 and the corresponding changes needed in EpiModel 2.0. Changes will be highlighted by comments.
+
+## SI Model with no vital dynamics:
+
+```{r echo = TRUE, eval = FALSE}
+## pre-EpiModel 2.0 Workflow
+num1 <- num2 <- 500
+nw <- network.initialize(n = num1 + num2, directed = FALSE, bipartite = num1)
+formation <- ~edges
+target.stats <- 400
+```
+
+```{r echo = TRUE, eval = FALSE}
+## EpiModel 2.0 
+num1 <- num2 <- 500
+
+#update to new initialization function network_initialize.
+nw <- network_initialize(n = num1 + num2)
+
+#Update to new vertex assignment function set_vertex_attribute
+nw <- set_vertex_attribute(nw, "group", rep(c(1,2), c(num1,num2)))
+
+#If looking to replicate 'bipartite' network:
+formation <- ~edges + nodematch("group")
+target.stats <- c(400, 0)
+```
+
+```{r echo = TRUE, eval = FALSE}
+coef.diss <- dissolution_coefs(dissolution = ~offset(edges), duration = 20)
+est <- netest(nw, formation, target.stats, coef.diss)
```

Comment:
Maybe add a comment to specify that this part is not changing. 

```suggestion
```{r echo = TRUE, eval = FALSE}
# Same for both workflow
coef.diss <- dissolution_coefs(dissolution = ~offset(edges), duration = 20)
est <- netest(nw, formation, target.stats, coef.diss)

Reply:
Overall, added more comments/verbose language explaining what changes and what stays the same.